task/07-yield-tasks.js

```diff
@@ -145,7 +204,17 @@ function* mergeSortedSequences(source1, source2) {
  *   Most popular implementation of the logic in npm https://www.npmjs.com/package/co
  */
 function async(generator) {
-    throw new Error('Not implemented');
+    const gen = generator();
+    
+    const handle = (res) => {
+        return (res.done) 
+                    ? Promise.resolve(res.value)
+                    : Promise.resolve(res.value).then((res) => {
+                                                        return handle(gen.next(res));
+                                                });
```

Comment:
avoid this code style, if-else is more readable than multiple conditional operators ? :